### PR TITLE
Fix receiving message from websocket client

### LINKED ISSUE #470: NIOWebSocketServer crash when sending message from JS to server
### Expected behavior
I expected to receive a message

### Actual behavior
I do not received a message

### Steps to reproduce

1. start web socket server with command eg swift run webSocketSwiftNIO 127.0.0.1 9889
2. connect to server from your browser in console 
var socket = new WebSocket("ws://127.0.0.1:9889/websocket");
or 
var socket = new WebSocket("ws://127.0.0.1:9889");
3. socket.send("111111");
4. crash server

### If possible, minimal yet complete reproducer code (or URL to code)
_[anything to help us reproducing the issue]_
ERROR:
Fatal error: tried to decode as type HTTPPart<HTTPRequestHead, ByteBuffer> but found IOData: file /webSocketSwiftNIO/.build/checkouts/swift-nio.git-3108475404973543938/Sources/NIO/NIOAny.swift, line 198

and method 
func channelRead(ctx: ChannelHandlerContext, data: NIOAny)

receives data
data: NIOAny(_storage: NIO.NIOAny._NIOAny.ioData(NIO.IOData.byteBuffer(ByteBuffer { readerIndex: 0, writerIndex: 12, readableBytes: 12, capacity: 1024, slice: _ByteBufferSlice(upperBound: 1024, _begin: 0), storage: 0x00007fd0af000000 (1024 bytes)})))

not    NIO.NIOAny._NIOAny.other( HTTPPart ...

### SwiftNIO version/commit hash
_[the SwiftNIO tag/commit hash]_
SwiftNIO 1.8.0

### Swift & OS version (output of `swift --version && uname -a`)
Apple Swift version 4.1.2 (swiftlang-902.0.54 clang-902.0.39.2)
Target: x86_64-apple-darwin17.5.0
Darwin Olehs-MacBook-Pro.local 17.5.0 Darwin Kernel Version 17.5.0: Fri Apr 13 19:32:32 PDT 2018; root:xnu-4570.51.2~1/RELEASE_X86_64 x86_64

### PR DESCRIPTION
**Pull request by woodcrust**
### Motivation:

I want to resolve #470 

### Modification:

remove httpHandler from pipeline
define example of method for printing received message to terminal

### Result:

Fix receiving message from websocket client

### CODE REVIEW

File: Sources/NIOWebSocketServer/main.swift [hunk 1]
```diff
@@ -203,16 +213,23 @@ let upgrader = WebSocketUpgrader(shouldUpgrade: { (head: HTTPRequestHead) in HTT
...
     .serverChannelOption(ChannelOptions.backlog, value: 256)
     .serverChannelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
 
     // Set the handlers that are applied to the accepted Channels
     .childChannelInitializer { channel in
-        let config: HTTPUpgradeConfiguration = (upgraders: [ upgrader ], completionHandler: { _ in })
+        let config: HTTPUpgradeConfiguration = (
+                        upgraders: [ upgrader ], 
+                        completionHandler: { _ in 
+                            _ = channel.pipeline.remove(handler: httpHandler)
```
Comment (Lukasa) [inline Sources/NIOWebSocketServer/main.swift]: I don’t think the return value should be suppressed here: the completion handler can handle future return values I think.

File: Sources/NIOWebSocketServer/main.swift [hunk 2]
```diff
@@ -122,6 +122,10 @@ private final class WebSocketTimeHandler: ChannelInboundHandler {
             self.pong(ctx: ctx, frame: frame)
         case .unknownControl, .unknownNonControl:
             self.closeOnError(ctx: ctx)
+        case .text:
+            var data = frame.unmaskedData
+            let text = data.readString(length: data.readableBytes) ?? ""
+            self.printToConsole(text: text)
```
Comment (Lukasa) [inline Sources/NIOWebSocketServer/main.swift]: I don’t think we need this abstraction: we can just call `print`.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

File: Sources/NIOWebSocketServer/main.swift [hunk 3]
```diff
@@ -203,16 +213,23 @@ let upgrader = WebSocketUpgrader(shouldUpgrade: { (head: HTTPRequestHead) in HTT
...
-        let config: HTTPUpgradeConfiguration = (upgraders: [ upgrader ], completionHandler: { _ in })
+        let config: HTTPUpgradeConfiguration = (
+                        upgraders: [ upgrader ], 
+                        completionHandler: { _ in 
+                            _ = channel.pipeline.remove(handler: httpHandler)
+                        }
+                    )
         return channel.pipeline.configureHTTPServerPipeline(withServerUpgrade: config).then {
-            channel.pipeline.add(handler: HTTPHandler())
+            channel.pipeline.add(handler: httpHandler)
```
Comment (normanmaurer) [inline Sources/NIOWebSocketServer/main.swift]: I think this is not correct as this way you will share the same `httpHandler` over multiple connections. I think this should be `HTTPHandler()`

File: Sources/NIOWebSocketServer/main.swift (hunk 2 above)
Comment (woodcrust) [reply Sources/NIOWebSocketServer/main.swift]: ok, I deleted printToConsole method

File: Sources/NIOWebSocketServer/main.swift (hunk 3 above)
Comment (woodcrust) [reply Sources/NIOWebSocketServer/main.swift]: I moved the definition of the httpHandler

File: Sources/NIOWebSocketServer/main.swift (hunk 1 above)
Comment (woodcrust) [reply Sources/NIOWebSocketServer/main.swift]: If you know how best to do this, then tell me, I'm happy to use it

File: Sources/NIOWebSocketServer/main.swift (hunk 3 above)
Comment (normanmaurer) [reply Sources/NIOWebSocketServer/main.swift]: @woodcrust that works....

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@normanmaurer Mind re-reviewing?

**Comment by normanmaurer** [comment]
LGTM... merged.

**Pull request merged**